lib/leaderboard.ts

```
@@ -0,0 +1,302 @@
+    const pitStats = new Map<
+      string,
+      { bouts: number; wins: number; votes: number }
+    >();
+    const votesByBout = new Map<string, Map<string, number>>();
+    const bestBoutByAgent = new Map<
+      string,
+      { boutId: string; votes: number }
+    >();
+
+    for (const vote of voteRows) {
+      const bout = boutMap.get(vote.boutId);
```

Biome (2.3.13)

[error] This variable is used before its declaration.

The variable is declared here:


(lint/correctness/noInvalidUseBeforeDeclaration)